src/_zkapauthorizer/tests/test_storage_protocol.py

```diff
@@ -1115,3 +1108,16 @@ def write_vector_to_read_vector(write_vector):
     write vector.
     """
     return (write_vector[0], len(write_vector[1]))
+
+
+def get_lease_grant_times(storage_server, storage_index):
+    """
+    Get the grant times for all of the leases for all of the shares at the
+    given storage index.
+    """
+    shares = storage_server._get_bucket_shares(storage_index)
```

Comment:
I'm curious about the use of this here vs. `get_all_share_paths` in `_storage_server.py`, and the new `get_share_path`. It seems like there are quite a few different places that track the on-disk layout.

Reply:
It's a mess.  I wouldn't mind having some time to do some cleanup, on the Tahoe-LAFS side as well as the ZKAPAuthorizer side.

Some of the inconsistency comes from changing my mind over time about how much work it is worth doing to avoid depending on yet another Tahoe-LAFS Python API by re-implementing something in ZKAPAuthorizer.  It would be nice to have a chance to make a pass over the codebase to bring it all into consistency with a single stance on this subject.